project/api/serializers.py

```diff
@@ -34,9 +35,44 @@
     Narrative,
     MapSettings,
     Narration,
+    NarrativeVote,
+    Profile,
 )
 
 
+class NarrativeVoteSerializer(ModelSerializer):
+    """
+    Serializes User votes for Narratives
+    """
+
+    def create(self, validated_data):
+        """
+        Updates NarrativeVote instance if it already exists
+        """
+
+        narrative_vote, _ = NarrativeVote.objects.update_or_create(
+            narrative=validated_data.get("narrative", None),
+            user=validated_data.get("user", None),
```

Comment:
That is dangerous, if you are authorized you can send any user_id you want using any id token.

Can you retrieve user data from request headers?
Anyway we don't have any mechanism to fetch user_id (for current user) from the backend, so we can't send this as a payload from frontend.